python/tox.ini

```diff
@@ -16,7 +16,7 @@
 # under the License.
 
 [tox]
-envlist = py37,py38,py39,linters
+envlist = py38,py39,linters
```

Comment:
1 year 4 months seems like quite a long time from now, do we really want to drop support?

Reply:
I agree, but Protocol was added in 3.8. If we're really aiming to use this for documentation purposes only and don't actually need the functionality of structural subtyping, I can change these to normal classes and include "These are Protocols for documentation purposes only" in the docstrings. How does that sound?

Reply:
Is the [backport](https://pypi.org/project/typing-extensions/) not useable to keep 3.7?

Reply:
You're absolutely right, I was getting an import error and thought it wasn't in the backport, but I was expecting an override of the normal typing import line. I'll update this to use `from typing_extensions` and add back 3.7. Thanks!

Reply:
Updated!